apps/photos/src/components/Collections/CollectionShare/emailShare.tsx

```diff
@@ -2,17 +2,49 @@ import SingleInputForm, {
     SingleInputFormProps,
 } from 'components/SingleInputForm';
 import { GalleryContext } from 'pages/gallery';
-import React, { useContext } from 'react';
+import React, { useContext, useState, useEffect } from 'react';
 import { t } from 'i18next';
 import { shareCollection } from 'services/collectionService';
 import { User } from 'types/user';
 import { handleSharingErrors } from 'utils/error/ui';
 import { getData, LS_KEYS } from 'utils/storage/localStorage';
 import { CollectionShareSharees } from './sharees';
+import { getLocalCollections } from 'services/collectionService';
 
 export default function EmailShare({ collection }) {
     const galleryContext = useContext(GalleryContext);
 
+    const [updatedOptionsList, setUpdatedOptionsList] = useState(['']);
+    let updatedList = [];
+    useEffect(() => {
+        const owner_user1: User = getData(LS_KEYS.USER);
+        const collection_list = getLocalCollections();
+        const getUpdatedOptionsList = async () => {
+            const result = await collection_list;
+            const emails = result.flatMap((item) => {
+                const shareeEmails = item.sharees.map((sharee) => sharee.email);
+                if (item.owner.email) {
+                    return [...shareeEmails, item.owner.email];
+                } else {
+                    return shareeEmails;
+                }
+            });
+
+            updatedList = Array.from(new Set(emails));
+
+            const filteredList = updatedList.filter(
+                (email) =>
+                    !collection.sharees
+                        .map((sharees) => sharees.email)
```

Comment:
we can move this outside the filter, that way we don't we won't have to map over it everytime

Reply:
Yes, I have moved it to a different variable (shareeEmailsCollection) outside the filter